spec/requests/api/v3/bikes_request_spec.rb

```diff
@@ -149,7 +149,132 @@
+             json_headers
+
+        bike2 = json_result["bike"]
+        expect(response.status).to eq(302)
+        expect(response.status_message).to eq("Found")
+        expect(bike1["id"]).to eq(bike2["id"])
+      end
+    end
+
+    context "given a bike with a match by a normalized email" do
+      it "responds with the match instead of creating a duplicate" do
+        bike_attrs = {
+          serial: "serial-01",
+          manufacturer: manufacturer.name,
+          color: color.name,
+          year: "1969",
+          owner_email: "bike-serial-01@example.com",
+        }
+        post "/api/v3/bikes?access_token=#{token.token}",
+             bike_attrs.to_json,
+             json_headers
+
+        expect(response.status).to eq(201)
+        expect(response.status_message).to eq("Created")
+        bike1 = json_result["bike"]
+
+        bike_attrs = bike_attrs.merge(owner_email: "  bike-serial-01@example.com  ")
+        post "/api/v3/bikes?access_token=#{token.token}",
+             bike_attrs.to_json,
+             json_headers
+
+        bike2 = json_result["bike"]
+        expect(response.status).to eq(302)
+        expect(response.status_message).to eq("Found")
+        expect(bike1["id"]).to eq(bike2["id"])
+      end
+    end
+
+    context "given a bike with a match by an owning user's secondary email" do
+      it "responds with the match instead of creating a duplicate" do
+        user.user_emails.create(email: "secondary-email@example.com")
+        bike = FactoryBot.create(:ownership, creator: user).bike
+
+        bike_attrs = {
+          serial: bike.serial,
+          manufacturer: bike.manufacturer.name,
+          color: color.name,
+          year: bike.year,
+          owner_email: user.secondary_emails.first,
+        }
+        post "/api/v3/bikes?access_token=#{token.token}",
+             bike_attrs.to_json,
+             json_headers
+
+        returned_bike = json_result["bike"]
+        expect(response.status).to eq(302)
+        expect(response.status_message).to eq("Found")
+        expect(returned_bike["id"]).to eq(bike.id)
+      end
+    end
+
+    context "given a bike with a match by serial" do
+      it "creates a new bike if the match has a different owner" do
+        bike = FactoryBot.create(:ownership, creator: user).bike
+
+        bike_attrs = {
+          serial: bike.serial,
+          manufacturer: bike.manufacturer.name,
+          color: color.name,
+          year: bike.year,
+          owner_email: "some-other-owner@example.com",
+        }
+        post "/api/v3/bikes?access_token=#{token.token}",
+             bike_attrs.to_json,
+             json_headers
+
+        returned_bike = json_result["bike"]
+        expect(response.status).to eq(201)
+        expect(response.status_message).to eq("Created")
+        expect(returned_bike["id"]).to_not eq(bike.id)
+      end
+
+      it "creates a new bike if it is POSTed by a creation org whose member initially created the match" do
```

Comment:
Is this statement accurate? Why does this create another bike?

Reply:
Is the test description a mistranslation of the following? 🤔 

> Bike was created by a user in an organization and has been claimed - creation organization posts a matching bike (what should happen?)

Reply:
> Why does this create another bike?

That's a good catch thanks -- it should be updating the ownership

Reply:
It didn't look like that ownership was claimed, so I was curious.

Reply:
@sethherr Re the claimed status -- I'm trying to figure out the intent of the existing logic here. Currently we reject updates if a bike is already claimed:

```rb
# app/controllers/api/v2/bikes.rb L175-176 (898bfe65)

@bike = found_bike
authorize_bike_for_user
```

```rb
# app/controllers/api/v2/bikes.rb L83-86 (00000000)

def authorize_bike_for_user(addendum = "")
  return true if @bike.authorize_for_user!(current_user)
  error!("You do not own that #{@bike.type}#{addendum}", 403)
end
```

```rb
# app/models/bike.rb L254-258 (3630a8d5)

def authorize_for_user!(u)
  return authorize_for_user(u) unless claimable_by?(u)
  current_ownership.mark_claimed
  true
end
```

```rb
# app/models/bike.rb L242-245 (aa4f5113)

def claimable_by?(u)
  return false if u.blank? || current_ownership.blank? || current_ownership.claimed?  <-----
  user == u || current_ownership.claimable_by?(u)
end
```

```rb
# app/models/bike.rb L247-251 (3630a8d5)

def authorize_for_user(u)
  return true if u == owner || claimable_by?(u)
  return false if u.blank? || current_ownership&.claimed    <-----
  authorized_by_organization?(u: u)
end
```

```
» claimable_by?(u)
false
» authorize_for_user(u)
false
```

In the preceding, the bike's `creation_organization` is the same as the current user's `organization`, and the bike is claimed:
```
» creation_organization.name
"Organization 1"
» u.organizations.first.name
"Organization 1"
» claimed?
true
```

Do we want to change this behavior? 

WIP implementation at 08badaba

Reply:
The intent of the existing behavior is that, until a user claims a bike, the organization that created it can continue to update it.

This behavior is to take care of these issues:
- A bike shop registers a bike for a user, but wants to add a photo after registering it
- ... Or some extra details about the bike (e.g. a specific component or a description)
- A bike was sent to the wrong email address, and the original owner needs to send it to the correct one

The initial goal was to make it so that the editing was unobtrusive, but cut off access to it once the user was officially in charge of the bike. This seemed like unsurprising behavior - since if you haven't officially claimed the bicycle, it seems reasonable that the person who sent it to you can update it. Once you've claimed it, you don't expect someone else to be able to go in and modify things about the bike.

When building this system, I imagined users would claim bikes within the next day and that organizations wouldn't have a desire to edit registrations once they had finished the initial editing of them. Both of these assumptions turned out to be incorrect - users wait longer and organizations sometimes want to update additional bits later on. I still think the current behaviour makes sense, but it doesn't cover all use cases.

Reply:
I reverted the test and implementation for that last user story — I think we should ship the bug fix and tests on this branch and spec out the remaining story separately to avoid scope creep.